src/SignalR/clients/java/signalr/src/main/java/com/microsoft/signalr/LongPollingTransport.java

```diff
@@ -0,0 +1,163 @@
+                this.headers.put("Authorization", "Bearer " + token);
+            }
+            return Single.just("");
+        });
+    }
+
+    @Override
+    public Completable start(String url) {
+        this.active = true;
+        logger.debug("Starting LongPolling transport.");
+        this.url = url;
+        pollUrl = url + "&_=" + System.currentTimeMillis();
+        logger.debug("Polling {}.", pollUrl);
+        return this.updateHeaderToken().flatMapCompletable((r) -> {
+            HttpRequest request = new HttpRequest();
+            request.addHeaders(headers);
+            return this.pollingClient.get(pollUrl, request).flatMapCompletable(response -> {
+                if (response.getStatusCode() != 200) {
+                    logger.error("Unexpected response code {}.", response.getStatusCode());
+                    this.active = false;
+                    return Completable.error(new Exception("Failed to connect."));
+                } else {
+                    this.active = true;
+                }
+                this.threadPool = Executors.newCachedThreadPool();
+                threadPool.execute(() -> poll(url).subscribeWith(receiveLoop));
+
+                return Completable.complete();
+            });
+        });
+    }
+
+    private Completable poll(String url) {
+        if (this.active) {
+            pollUrl = url + "&_=" + System.currentTimeMillis();
+            logger.debug("Polling {}", pollUrl);
+            return this.updateHeaderToken().flatMapCompletable((x) -> {
+                HttpRequest request = new HttpRequest();
+                request.addHeaders(headers);
+                Completable pollingCompletable = this.pollingClient.get(pollUrl, request).flatMapCompletable(response -> {
+                    if (response.getStatusCode() == 204) {
+                        logger.info("LongPolling transport terminated by server.");
+                        this.active = false;
+                    } else if (response.getStatusCode() != 200) {
+                        logger.error("Unexpected response code {}.", response.getStatusCode());
+                        this.active = false;
+                        this.closeError = "Unexpected response code " + response.getStatusCode() + ".";
+                    } else {
+                        if (response.getContent() != null) {
+                            logger.debug("Message received.");
+                            threadPool.execute(() -> this.onReceive(response.getContent()));
+                        } else {
+                            logger.debug("Poll timed out, reissuing.");
+                        }
+                    }
+                    return poll(url);
+                });
+
+                return pollingCompletable;
+            });
+        } else {
+            logger.debug("Long Polling transport polling complete.");
+            receiveLoop.onComplete();
+            if (!stopCalled) {
```

Comment:
This isn't being set anywhere.
A bool isn't very thread safe, might want to consider an atomic or something else.
Add a test that the 204 calls onclose and afterwards stop doesn't.
And another for calling stop and not getting 2 onclose callbacks.